[release] v9.5.0 (#22742)
mui/mui-x: CHANGELOG.md

# Changelog

## 9.5.0

> <!-- generated comparing v9.4.0..master -->

Reviewer: ```suggestion
```

> _Jun 10, 2026_

Reviewer: ```suggestion
_Jun 11, 2026_
```

> We'd like to extend a big thank you to the 7 contributors who made this release possible. Here are some highlights ✨:

Reviewer: ```suggestion
We'd like to extend a big thank you to the 8 contributors who made this release possible. Here are some highlights ✨:
```

> - 🎁 New [`multiSelect` column type for Data Grid Pro](https://mui.com/x/react-data-grid/column-definition/#MultiSelectColumn)

Reviewer 2: ```suggestion
- 🎁 New [`multiSelect` column type for Data Grid Pro](https://mui.com/x/react-data-grid/column-definition/#MultiSelectColumn)
- 🚀 New [renderer="svg-progressive"](https://mui.com/x/react-charts/scatter/#progressive-renderer) paints scatter points in batched frames so large datasets render smoothly without blocking the main thread.
```

- 🐞 Bugfixes
- 📚 Documentation improvements

> Special thanks go out to community member @mustafajw07 for their valuable contribution.

Reviewer 3: ```suggestion
Special thanks go out to community members @mustafajw07 and @Anexus5919 for their valuable contribution.
```
After merging this [PR](https://github.com/mui/mui-x/pull/22552) we should add Anexus5919 here too 🙌

Author: Yes, I'll take care of the changelog

Special thanks go out to community member @mustafajw07 for their valuable contribution.

The following team members contributed to this release:

> @alexfauquette, @JCQuintas, @rita-codes, @rluzists1, @sai6855, @siriwatknp

Reviewer: ```suggestion
@alexfauquette, @arminmeh, @JCQuintas, @rita-codes, @rluzists1, @sai6855, @siriwatknp
```

### Data Grid

#### `@mui/x-data-grid@9.5.0`

> - [DataGrid] Restrict warning messages to non-production environments in various components (#22461) @sai6855

Reviewer: ```suggestion
- [DataGrid] Restrict warning messages to non-production environments in various components (#22461) @sai6855
- [DataGrid] Prevent React state update before mount (#22374) @arminmeh
```

- [charts] Document focus indicator in composition page (#22712) @JCQuintas
- [charts] Fix `ChartsTooltip` `container` and `disablePortal` props being ignored (#22690) @JCQuintas
- [charts] Make ordinal scales O(1) to copy and re-range (#22691) @JCQuintas
- [charts] Stabilize WebGL bar rendering at sub-pixel widths (#22678) @JCQuintas
- [charts] Use the series config to define value types (#22693) @sai6855
- [charts] Decouple interaction hook from the cartesian zoom (#22708) @alexfauquette

#### `@mui/x-charts-pro@9.5.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@9.5.0`, plus:

- [charts-pro] Add missing focuse highlight (#22689) @alexfauquette

#### `@mui/x-charts-premium@9.5.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-charts-pro@9.5.0`, plus:

- [charts-premium] Set `Heatmap` WebGL render as stable (#22665) @alexfauquette
- [charts-premium] Stabilize candle stick charts (#22666) @alexfauquette
- [charts-premium] Stabilize the radial charts (#22655) @alexfauquette

### Scheduler

> #### `@mui/x-scheduler@9.0.0-beta.0`

Reviewer: ```suggestion
#### `@mui/x-scheduler@9.0.0-beta.1`
```

> - [scheduler] Remove dead CSS variable writes in `EventTimelinePremiumContent` (#22720) @rita-codes

Reviewer: I guess the versions in the changelog got messed up, because this only affects premium. Premium is the only version that got bumped.

Author: Yeah, the premium package is the only one that was updated since the previous release.
I'll clean the changelog to only mention scheduler premium

Author: Or should I skip scheduler completely? @mui/scheduler

Reviewer 3: I'm merging this PR (https://github.com/mui/mui-x/pull/22552) that also affects the premium version, do you mind waiting for it? 🙏 so we can have both and the changelog will look better :D

Author: Sure, go ahead 👍🏻

Reviewer 3: merged!

> - [scheduler] Remove dead CSS variable writes in `EventTimelinePremiumContent` (#22720) @rita-codes

Reviewer 3: ```suggestion
- [scheduler] Remove dead CSS variable writes in `EventTimelinePremiumContent` (#22720) @rita-codes
- [scheduler] Show scope dialog when deleting a recurring event (#22552) @Anexus5919
```

Author: I've bumped other scheduler packages too (except 2 headless ones). Is this correct?

Reviewer 3: Yesss, we had to change a small thing there, so that's expected (we updated the headless packages names, now we call them internals)

Reviewer 3: But the change should be under premium, because the recurring events are only available on premium, so 
you will never see the scope dialog on the community version

Reviewer 3: The change we made on the community version is just a rename of the function that triggers the deletion (and decides if we show or not the scope dialog)

Author: OK, moved it to premium
